components/TodoForm.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useRef, useContext } from "react";
+import TodoModel from "../models/todo";
+import { TodoContext } from "../store/store-todo";
+import classes from "../styles/Home.module.css";
+
+const TodoForm = () => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const todoCtx = useContext(TodoContext);
+  const addTodo = todoCtx.addTodo;
+
+  const submitHandler = (e: React.FormEvent) => {
+    e.preventDefault();
+    const newTodo: TodoModel = {
+      id: new Date().getTime().toString(),
+      title: inputRef.current!.value,
+      completed: false,
+    };
+
+    if (newTodo.title.trim() === "") {
+      return;
+    }
+    addTodo(newTodo);
+
+    inputRef.current!.value = "";
```

Comment:
(!.) <-- what is it used for?

Reply:
> (!.) <-- what is it used for?

Non-null assertion operator
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator